src/gpu/generic/sycl/ref_resampling.hpp

```diff
@@ -47,21 +47,13 @@ struct ref_resampling_fwd_t : public gpu::generic::sycl::primitive_t {
             const memory_desc_wrapper src_d(src_md(0));
             const memory_desc_wrapper dst_d(dst_md(0));
 
-            const bool ok = src_md()->data_type == dst_md()->data_type
-                    && set_default_params() == status::success
+            const bool ok = is_fwd() && set_default_params() == status::success
```

Comment:
`set_default_params() == status::success` is quite expensive check. You want it to be the pre-last one following by checks relying on the formats set.

Reply:
I did move that call following your suggestion @dzarukin . Thank you. Let me know if something else is needed with this PR.